### PR TITLE
[INTERNAL] CONTRIBUTING.md: Add section about multi branch development

### PR DESCRIPTION
**Pull request by flovogt**
JIRA: CPOUI5FOUNDATION-775

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
+There are phases when UI5 Tooling has already started developing a new major versions of it's projects but does not has released the new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current and the new major version.
+
```
Comment (RandomByte) [inline CONTRIBUTING.md:121]: I would really like a short example here. Something like "Current version : v3; New Major: v4".

Otherwise looks great!

Comment (flovogt) [reply CONTRIBUTING.md:121]: done. If you're fine with the current state. I will assign UA.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
+There are phases when UI5 Tooling has already started developing a new major version of it's projects but does not has released the new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```
Comment (d3xter666) [inline CONTRIBUTING.md:120]:
```suggestion
There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't released a new major version. This sitatuation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
+There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't yet released a new major version. This situation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
+
+1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or [INTERNAL]
+2. Rebase and merge the PR in the current version
+3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change appears in the changelog of the new major version
```
Comment (KlattG) [inline CONTRIBUTING.md]:
```suggestion
3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change does not appear in the changelog of the new major version
```

**Review by KlattG — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (flovogt) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
+There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't yet released a new major version. This situation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```
Comment (KlattG) [inline CONTRIBUTING.md:120]:
```suggestion
There are phases when development of a new major version of UI5 Tooling has already started, but this new major version hasn't been released yet. This situation requires a special handling of pull requests / commits if the actual coding change should be both part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
...
+
+1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or `[INTERNAL]`
+2. Rebase and merge the PR in the current version
+3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this it is ensured the coding change does not appear in the changelog of the new major version
```
Comment (KlattG) [inline CONTRIBUTING.md:124]:
```suggestion
3. Cherry-pick this commit to the new major version (e.g. Version 4) and change the commit message header to `[INTERNAL]`. With this, you ensure that the coding change does not appear in the changelog of the new major version.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
+There are phases when UI5 Tooling has already started developing a new major version of its projects but haven't yet released a new major version. This situation requires a special handling of pull requests / commits if the actual coding change should be part of the current (e.g. Version 3) and the new major version (e.g. Version 4).
+
+1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or `[INTERNAL]`
```
Comment (KlattG) [inline CONTRIBUTING.md:122]:
```suggestion
1. Create a pull request for the current version (e.g. Version 3) with the desired commit message header `[FEATURE]`, `[FIX]` or `[INTERNAL]`.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
...
+2. Rebase and merge the PR in the current version
```
Comment (KlattG) [inline CONTRIBUTING.md:123]:
```suggestion
2. Rebase and merge the PR in the current version.
```

**Review by KlattG — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -115,3 +115,10 @@ This DCO replaces the previously used CLA ("Contributor License Agreement") as w
     - Note that the UI5 developers have many duties. So, depending on the required effort for reviewing, testing, and clarification, this may take a while.
 1. Once the change has been approved and merged, we will inform you in a comment.
 1. Celebrate! 🎉
+
+### Multi Branch Development
```
Comment (KlattG) [inline CONTRIBUTING.md:119]:
```suggestion
### Multi-Branch Development
```

**Review by KlattG — COMMENTED** [review]
Just one tiny little thing, otherwise LGTM

**Review by KlattG — APPROVED** [review]
LGTM

**Review by RandomByte — APPROVED** [review]

**Pull request merged**